### PR TITLE
OS X launchd jobs for cron

### PR DESCRIPTION
**Pull request by wadebenson**
I wrote this so thought I would contribute them back.

### CODE REVIEW

**Comment by juho-jaakkola** [comment]
I'm not that familiar with OS X. Is this the default way of configuring cron jobs for it? Doesn't it have command line tool like [cron](https://en.wikipedia.org/wiki/Cron)?

**Comment by juho-jaakkola** [comment]
I should had Googled before asking :)

> Although it is still supported, cron is not a recommended solution. It has been deprecated in favor of launchd.

**Comment by sh3llc0de** [comment]
Indeed.  It is possible to include all the intervals into one launchd task.. I haven't found a way to hit 10 different url's from one launchd task.

We could introduce one url for cron that gets called all the time.  Where all the cron jobs just keep a timestamp and only fire once it's window threshold has exceeded. (i.e. 1min, 5min, 15min, ...)

**Comment by juho-jaakkola** [comment]
@mrclay I believe you are also using OS X, so can you do a review for this?

**Comment by juho-jaakkola** [comment]
@sh3llc0de in the mean time you can:
 - Squash the three separate commits into a single one
 - Update the commit message to use our [message format](http://learn.elgg.org/en/1.12/contribute/code.html?highlight=rebase#commit-message-format)

See the [rebase](http://learn.elgg.org/en/1.12/contribute/code.html?highlight=rebase#rewriting-commit-messages) part for instructions.

But don't worry if that sounds complicated. We can take care of it if necessary.

**Comment by sh3llc0de** [comment]
I'm very familiar with git no problem.. (just first submitting to Elgg)

File: engine/lib/cron.php [hunk 1]
```diff
@@ -35,6 +35,49 @@ function _elgg_cron_init() {
 }
 
 /**
+ * Cron run handler
+ *
```
Comment (juho-jaakkola) [inline engine/lib/cron.php]: Could you explain in the docblock, what this function is doing and why it is needed (as opposed to the other approach)?

Format:
```php
/**
 * First line
 * 
 * One or more lines, max 80 characters each. Lorem ipsum dolor sit
 * amet.
 * 
 * @access private
 */
```

**Comment by sh3llc0de** [comment]
Ok all comments added.

**Comment by juho-jaakkola** [comment]
I apologize for being such a pain in the ass, but could you also change the PR for the 2.x branch instead of the master branch?

Bear with me! You're doing great! :)

**Comment by sh3llc0de** [comment]
I don't mind.. git handles this stuff pretty straight forward.

Ok done.. put onto branch 2.x

**Comment by juho-jaakkola** [comment]
I meant that you need to close this PR and open a new one that points to the 2.x branch.

**Comment by sh3llc0de** [comment]
I don't understand.. I am new to github... 

This PR is rebased to Branch 2.x, is it because of how PR's work with github?

**Comment by juho-jaakkola** [comment]
Yeah, the target branch doesn't change automatically. Additionally you cannot change the target branch of an exiting PR, but you need to open a new one.

**Comment by juho-jaakkola** [comment]
I've closed this one. Now just got to "Pull requests" page and open a new one target for the 2.x branch.

**Comment by sh3llc0de** [comment]
Already done.. np. thanks for the help.

**Pull request closed without merging**